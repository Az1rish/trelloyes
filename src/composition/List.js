import React from 'react';
import Card from './Card';
import STORE from '../store';

function List(props) {
    const arrayOfCard = STORE.allCards.map((card) =>
        <Card title={card.title} content={card.content} />
    )
    const arrayOfLis = STORE.lists.map((list) =>
    <section className="List" key={list.id}>
        <header className="List-header">
            <h2>{list.header}</h2>
        </header>
        <div className="List-cards">
            {arrayOfCard}
        </div>
    </section>
  )
    return (
        <div className="App-list">
            {arrayOfLis}   
        </div> 
    )
}

export default List;