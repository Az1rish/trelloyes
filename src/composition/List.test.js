import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('Card component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDom.render(<List />, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const cards = [ {id: 'l', title: 'm', content: 'content'} ];
        const tree = renderer
            .create(<List key="Key" header="Title" cards={cards} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
