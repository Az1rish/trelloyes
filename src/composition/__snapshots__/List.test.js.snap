// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Title
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        m
      </h3>
      <p>
        content
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
